chains/btc/uploader/ipfs.go

```diff
@@ -0,0 +1,81 @@
+package uploader
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"mime/multipart"
+	"net/http"
+
+	"github.com/ChainSafe/sygma-relayer/chains/btc/config"
+)
+
+type Uploader interface {
+	Upload(proposals []map[string]interface{}) (string, error)
+}
+
+type IPFSUploader struct {
+	config config.UploaderConfig
+}
+
+func NewIPFSUploader(config config.UploaderConfig) *IPFSUploader {
+	return &IPFSUploader{config: config}
+}
+
+type IPFSResponse struct {
+	IpfsHash string `json:"IpfsHash"`
+}
+
+func (s *IPFSUploader) Upload(dataToUpload []map[string]interface{}) (string, error) {
+
+	// Convert proposals to JSON
+	jsonData, err := json.Marshal(dataToUpload)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a multipart form file
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", "metadata.json")
```

Comment:
Does the file name matter?
Could this overwrite previous pins?

Reply:
I tested and it didn't overwrite